GliaWidgetsTests/Sources/InteractorTests.swift

```diff
@@ -423,6 +423,32 @@ class InteractorTests: XCTestCase {
         XCTAssertEqual(callbacks, [.sendMessageWithAttachment])
     }
 
+    func test_sendMessageFailsWith401Error() throws {
+        var interactorEnv = Interactor.Environment.failing
+        let expectedError: CoreSdkClient.GliaCoreError = .init(
+            reason: "Expired access token",
+            error: CoreSdkClient.Authentication.Error.expiredAccessToken
+        )
+        interactorEnv.coreSdk.sendMessageWithMessagePayload = { payload, result in
+            result(.failure(expectedError))
+        }
+        let interactor = Interactor.mock(environment: interactorEnv)
+
+        interactor.send(messagePayload: .mock(content: "mock-message")) { result in
+            switch result {
+            case .success:
+                XCTFail()
+            case let .failure(error):
+                switch error.error {
```

Comment:
Since this assertion happens in callback, there's potential probability that this callback will be never called or called later where test already has finished its execution. I would suggest to store result in the array declared on line 436, the assert contents of array out of callback. This will assert that callback is called expected number of times, and also that result is also expected.

Reply:
Added callbacks to improve the test [a9ed2df](https://github.com/salemove/ios-sdk-widgets/pull/959/commits/a9ed2dfd0e961615034cc77c5e4e3e6c682bc6d5)